### PR TITLE
feat(subscription): support oidcToken

### PR DESCRIPTION
**Pull request by stayradiated**
### Issue

It is currently not possible to create a new Subscription with the **Service Account Email** or **Audience** properties configured. These properties are part of the `PushConfig.oidcToken` field which is not being formatted correctly.

### Changes

Similar to the `pushEndpoint` option, when formatting the metadata of a Subscription, the property needs to be moved into the `pushConfig` object.

```typescript
    if (metadata.pushEndpoint) {
      formatted.pushConfig = {
        pushEndpoint: metadata.pushEndpoint,
      };
      delete formatted.pushEndpoint;
    }

    // fix
    if (metadata.oidcToken) {
      formatted.pushConfig = {
        ...formatted.pushConfig,
        oidcToken: metadata.oidcToken,
      };
      delete formatted.oidcToken;
    }
```

I have added tests and updated documentation for `PubSub.createSubscription` and `Subscription.modifyPushConfig`.

### CODE REVIEW

**Comment by feywind** [comment]
@bcoe Code review wise this looks fine to me, but I don't know all the inner workings enough to give a yay/nay on the functionality. Do you mind taking a quick look tomorrow? (Or tag someone who would know...)

**Comment by bcoe** [comment]
@feywind this looks reasonable to me too, but I think perhaps we should loop in @bshaffer for a review who's been doing work around ID Tokens.

**Comment by bcoe** [comment]
@stayradiated have you tested end to end with your branch against the actual service?

File: src/subscription.ts [hunk 1]
```diff
@@ -1039,6 +1050,14 @@ export class Subscription extends EventEmitter {
       delete formatted.pushEndpoint;
     }
 
+    if (metadata.oidcToken) {
+      formatted.pushConfig = {
+        ...formatted.pushConfig,
+        oidcToken: metadata.oidcToken,
+      };
+      delete formatted.oidcToken;
```
Comment (bcoe) [inline src/subscription.ts]: I'm relatively green in this codebase, but mind clarifying why we need to delete the key off the `formatted` object?

Comment (bshaffer) [reply src/subscription.ts]: It does seem to already be happening above it for `formatted.pushEndpoint` and as well...

Comment (feywind) [reply src/subscription.ts]: Yeah, that was my take as well, oidcToken is just being treated like what's above. My main question is on the sort of systemic goodness of the patch - I don't understand the gestalt of the library enough to say yay or nay yet.

Comment (bshaffer) [reply src/subscription.ts]: I believe the changes here just expose access to the underlying API call, and is already exposed in other client libraries. With the exception of deleting the key (is this for memory management?) this is LGTM.

**Pull request merged**